dask_glm/normal.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import absolute_import, division, print_function
+
+from dask_glm.utils import dot
+
+
+try:
+    from numba import jit
+except ImportError:
+    def jit(*args, **kwargs):
+        def _(func):
+            return func
+        return _
+
+
+def loglike(Xbeta, y):
+    return ((y - Xbeta)**2).sum()
+
+
+def pointwise_loss(beta, X, y):
+    beta, y = beta.ravel(), y.ravel()
+    Xbeta = X.dot(beta)
+    return loglike(Xbeta, y)
+
+
+def pointwise_gradient(beta, X, y):
+    beta, y = beta.ravel(), y.ravel()
+    Xbeta = X.dot(beta)
+    return gradient(Xbeta, X, y)
+
+
+def gradient(Xbeta, X, y):
+    return 2 *  dot(X.T, Xbeta) - 2 * dot(X.T, y)
+
+
+def hessian(Xbeta, X):
+    return dot(X.T, X)
```

Comment:
In conversation I originally pushed for functions like newton, gradient_descent, etc.. to just take in flat functions.  At the time I didn't realize how similarly structured the different approaches here would be.  I'd like to relax my request to just that those functions should take in something simple, that is hopefully separated from other concerns like "should we use the statsmodels or sklearn APIs".  It may be that constructing a simple class around these functions is best.  

Three options come to mind:

1.  A class with these functions as static methods.  We would never instantiate the class, but just use it as a namespace
2.  A class with these functions as methods.  This would be useful if future options other than Linear and Logistic regression are likely to have parameters that we would want to tune
3.  Leave it as is, but use the modules themselves as inputs
4.  Leave it as is, and continue passing in functions directly